exts/cesium.omniverse/cesium/omniverse/ui/credits_viewport_frame.py

```diff
@@ -52,41 +52,47 @@ def _on_update_frame(self, _e: carb.events.IEvent):
         if self._data_attribution_button is None:
             return
 
-        credits_available = self._cesium_omniverse_interface.credits_available()
+        new_credits = self._cesium_omniverse_interface.get_credits()
 
-        if credits_available != self._data_attribution_button.visible:
-            if credits_available:
+        if new_credits != self._credits:
+            self._credits.clear()
+            self._credits.extend(new_credits)
+            self._build_fn()
+
+        has_offscreen_credits = False
+        for _, show_on_screen in new_credits:
+            if not show_on_screen:
+                has_offscreen_credits = True
+
+        if has_offscreen_credits != self._data_attribution_button.visible:
+            if has_offscreen_credits:
                 self._logger.info("Show Data Attribution")
             else:
                 self._logger.info("Hide Data Attribution")
-            self._data_attribution_button.visible = credits_available
-
-        if self._data_attribution_button.visible:
-            new_credits = self._cesium_omniverse_interface.get_credits()
-            if new_credits is not None and len(self._credits) != len(new_credits):
-                self._credits.clear()
-                self._credits.extend(new_credits)
-                self._build_fn()
-        else:
-            self._credits.clear()
-            self._build_fn()
+            self._data_attribution_button.visible = has_offscreen_credits
+
+        self._cesium_omniverse_interface.credits_start_next_frame()
 
     def _on_data_attribution_button_clicked(self):
-        self._credits_window = CesiumOmniverseCreditsWindow(self._cesium_omniverse_interface)
+        self._credits_window = CesiumOmniverseCreditsWindow(self._cesium_omniverse_interface, self._credits)
 
     def _build_fn(self):
         with self._credits_viewport_frame:
             with ui.VStack():
                 ui.Spacer()
                 with ui.HStack(height=0):
-                    ui.Spacer()
-
-                    CesiumCreditsParser(self._credits, should_show_on_screen=True)
-
-                    self._data_attribution_button = ui.Button(
-                        "Data Attribution",
-                        visible=False,
-                        width=0,
-                        height=0,
-                        clicked_fn=self._on_data_attribution_button_clicked,
-                    )
+                    with ui.HStack(height=0, spacing=4):
```

Comment:
This `HStack` needs a maximum width. The algorithm I came up with yesterday is one way to do this based on the viewport width. Right now it's bleeding into the axis indicator which is a very bad UX.